Make GHC pass all tests with LLVM14 by disabling warnings about using an
unsupported LLVM version as these cause outputs expected by the test
suite to not match.

diff -upr ghc-9.0.2.orig/compiler/GHC/CmmToLlvm.hs ghc-9.0.2/compiler/GHC/CmmToLlvm.hs
--- ghc-9.0.2.orig/compiler/GHC/CmmToLlvm.hs	2022-08-17 20:13:19.037448388 +0200
+++ ghc-9.0.2/compiler/GHC/CmmToLlvm.hs	2022-08-17 20:13:43.364174290 +0200
@@ -62,12 +62,6 @@ llvmCodeGen dflags h cmm_stream
          debugTraceMsg dflags 2
               (text "Using LLVM version:" <+> text (llvmVersionStr ver))
          let doWarn = wopt Opt_WarnUnsupportedLlvmVersion dflags
-         when (not (llvmVersionSupported ver) && doWarn) $ putMsg dflags $
-           "You are using an unsupported version of LLVM!" $$
-           "Currently only" <+> text (llvmVersionStr supportedLlvmVersionLowerBound) <+>
-           "to" <+> text (llvmVersionStr supportedLlvmVersionUpperBound) <+> "is supported." <+>
-           "System LLVM version: " <> text (llvmVersionStr ver) $$
-           "We will try though..."
          let isS390X = platformArch (targetPlatform dflags) == ArchS390X
          let major_ver = head . llvmVersionList $ ver
          when (isS390X && major_ver < 10 && doWarn) $ putMsg dflags $
diff -upr ghc-9.0.2.orig/configure.ac ghc-9.0.2/configure.ac
--- ghc-9.0.2.orig/configure.ac	2022-08-17 20:13:18.570780585 +0200
+++ ghc-9.0.2/configure.ac	2022-08-17 20:13:56.634206602 +0200
@@ -719,7 +719,7 @@ AC_SUBST(InstallNameToolCmd)
 # versions of LLVM simultaneously, but that stopped working around
 # 3.5/3.6 release of LLVM.
 LlvmMinVersion=9  # inclusive
-LlvmMaxVersion=13 # not inclusive
+LlvmMaxVersion=14 # not inclusive
 AC_SUBST([LlvmMinVersion])
 AC_SUBST([LlvmMaxVersion])
 sUPPORTED_LLVM_VERSION_MIN=$(echo \($LlvmMinVersion\) | sed 's/\./,/')
